src/Stream.re

```diff
@@ -1,74 +1,53 @@
-type sendFunc('a) = 'a => unit;
-/* type complete = unit => unit; */
-
-type streamFunc('a) = sendFunc('a) => unit;
-
-type unsubscribeFunc = unit => unit;
-
-type subscriber('a) = {
+type subscriber('msg) = {
   id: int,
-  onValue: sendFunc('a),
+  onValue: 'msg => unit,
   /* onComplete: complete, */
 };
 
-type t('a) = {
+type t('msg) = {
   lastSubscriberId: ref(int),
-  subscribers: ref(list(subscriber('a))),
+  subscribers: ref(list(subscriber('msg))),
 };
 
-let ofDispatch = (f: streamFunc('a)) => {
-  let subscribers: ref(list(subscriber('a))) = ref([]);
+let ofDispatch = f => {
```

Comment:
This name is confusing. `f` is not a dispatch, so it is in fact not "of dispatch", but " of _something_ that is given a dispatch function"

Reply:
Totally, this API is pretty confusing. I'd like to just remove it completely from this library - I'm not sure it is necessary (besides supporting some code in Onivim 2 that still uses it..).

Reply:
And remove the only internal usage of `Isolinear.Stream.map`. 

The only place I could find is here:
https://github.com/onivim/oni2/blob/50f6d440b126f60d0d8d6cde793513c5595d1e9a/src/Store/StoreThread.re#L266

If we can get rid of that wart, I think we can remove `map` and `ofDispatch`